src/main/java/org/mifos/connector/channel/camel/routes/ChannelRouteBuilder.java

```diff
@@ -270,10 +270,28 @@ public void configure() {
                     }
                     extraVariables.put(TENANT_ID, tenantId);
 
+                    String channelRequestBodyString = exchange.getIn().getBody(String.class);
+                    JSONObject body = new JSONObject(channelRequestBodyString);
+                    JSONArray payer = body.getJSONArray("payer");
+                    String phoneNumber = "";
+                    String  accountId = "";
+                    if ( ((JSONObject) payer.get(0)).getString("key") == "key") {
+                        phoneNumber = ((JSONObject) payer.get(0)).getString("value");
+                        accountId = ((JSONObject) payer.get(1)).getString("value");
```

Comment:
@danishjamal104 : How are you sure about the array ordering? As per API spec, an API consumer can pass accountId as the first key:value pair in the array and phoneNumber as the 2nd key value pair in the array.

Reply:
updated this, please check now. Thanks :)